app/src/main/java/lt/vilnius/tvarkau/SignInActivity.java

```diff
@@ -0,0 +1,104 @@
+package lt.vilnius.tvarkau;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.View;
+import android.widget.Toast;
+
+import com.google.android.gms.auth.api.Auth;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.auth.api.signin.GoogleSignInResult;
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+
+/*
+    Activity which handles log in of user.
+    It does not matter whether user decides to do it,
+    he will be transferred to MainActivity
+
+ */
+public class SignInActivity extends AppCompatActivity implements GoogleApiClient.OnConnectionFailedListener{
```

Comment:
Well, seems like we have code duplication here. Not good. Maybe let's extract everything g+ signin related to another class and make composition instead of code duplication?